star trails

gzotti opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->

### Description
<!--- Please include a summary of the change and which issue is fixed. -->
<!--- Please also include relevant motivation and context. -->
<!--- List any dependencies that are required for this change. -->

This little branch adds an interesting "wow" effect by not blanking frames. You can configure a keyboard shortcut to toggle this, and then render star trails.

This is a bit "hacky"! One problem are extended objects: Zodiacal light, Milky Way, atmosphere, DSO images need to be switched off first. This is handled automatically and easy to solve.

The worst problem is that the scene becomes too bright too quickly when any extended objects would be plotted.  A solution would be to not just "forget" blanking, but add a slight dimming of the scene. This should then result in an afterglow effect where star trails don't get infinitely long. Maybe the dimming can be made configurable. 

Due to the nature of this method, screenshots have to be made with operating system tools, not Stellarium's own screenshot renderer.

OpenGL/Qt experts please continue on this. Change StelCore::preDraw() and implement the methods in the StelViewportFaderEffect class, or maybe directly in StelCore::preDraw().

### Screenshots (if appropriate):
![Screenshot (221) (Klein)](https://user-images.githubusercontent.com/18099873/133320943-edf1c95a-838a-42d6-94ff-a668fa9fcddb.png)

### Type of change
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] This change requires a documentation update

### How Has This Been Tested?
<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->

**Test Configuration**:
* Operating system: Windows 10 21H1; Windows 11 25H2
* Graphics Card: Nvidia Geforce 960M; Geforce RTX 3070Ti
 
## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] My code follows the [code style](http://stellarium.org/doc/head/codingStyle.html) of this project.
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

gzotti committed:
Attempt creating a star trail effect.
Currently this leads to a white-out within seconds.
We would need an fps-dependent frame decay.

guillaumechereau commented:
I tried to complete the viewport effect code, using a blending function with alpha = 0.08:
    	 GL_CONSTANT_ALPHA, GL_ONE_MINUS_CONSTANT_ALPHA

I can get this result:

![image](https://user-images.githubusercontent.com/107679/139009820-cbf67502-0112-4bae-ad12-62e5dc0dc10a.png)

While the trails are not very clear, this has the nice effect of not affecting the non moving parts of the screen, like the landscape.

Should I commit in this branch?

gzotti commented:
Yes, this looks very good for a start, thank you. I may be then able to play around with different alphas and maybe still different blending functions, and resetting the screen when view direction was changed etc. The non-affection of the landscape is very welcome!

guillaumechereau commented:
OK I'll commit what I have, feel free to modify it.  Unfortunately the only way to keep the landscape stable is to use a blending function of the form:    color = new * t + prev * (1 - t), and instead of the accumulating effect, we get something that looks more like a blur.  Not sure how to improve that.

The other problem here is that we apply the effect to everything, including the labels and UI, so it makes the interface clumsy.

gzotti commented:
Thank you, it is certainly a progress. But the blur is strange, indeed.  

I think we can auto-hide the GUI while the effect is active, or do similar things. It's mostly a show-off effect.

I wonder how alterBuffer would ever be applied in a ViewportEffect?

luzpaz commented:
Bump

gzotti committed:
Suppress drawing of overly bright scene elements in "starstreak mode"

gzotti committed:
Inhibit planet hint&labels
Inhibit star labels
Inhibit Gridlines and Nebula hints/labels
Reactivate Landscape

gzotti commented:
Some modules are auto-inhibited now: gridlines, labels, planet and DSO hints, planet halos, Milky Way, Zodiacal light. 

Now it depends on what users expect. 
The best use is to switch off landscape and press Ctrl-Alt-C. This works for a few spectacular effects in a demo, but can run into really ugly white-outs if not applied with caution. 

I would like to see landscape being replotted and NOT run into white-out, however I am unable to see an easy solution. 

Also, changing view direction or fov changes should re-initialize (clear) the view. This would need a few new signals. Is the overhead worth it?

I cannot get @guillaumechereau 's solution to show again. :-(

10110111 commented:
Why not just draw all the cosmic items into a separate FBO for accumulation, and blit on each frame into the main render target after rendering into this FBO?

gzotti commented:
Sounds like a plan :-) 
I think by now you have the better overview about which buffers exist and how they correlate with each other. (This architecture should be described somewhere!!!) It may also be related to your longer-term ideas around using physical radiance units and post-process effect filtering (blur up bright stars/planets) which should also be applied to the sky only. (And not on labels, hint icons, ... or landscapes.) I will be glad if you bring in your skills here.

The net view effect I envision in this mode should be "interactive foreground" (i.e. landscape should redraw well), stars should not run over the landscape (I tried several blending options, all but current looked bad), and on change of view direction, this "sky buffer" should reset to black (or background color). Dithering should be applied only to the final view (Currently it of course accumulates and looks bad as well). An extension could be that the buffer is dimmed instead of blanked, so that the star trails fade away after e.g. 10...60 seconds. 

If possible, gridlines or DSO markers, or even the MilkyWay/ZL could be made visible again without streaking. (I.e., not put into that "celestial" FBO. (But another FBO for MW/ZL/DSO textures/DSS/HiPS is yet another indirection, may be overkill, so IMHO it's OK to just switch them off). 


This all is meant to be for stronger systems. The "low graphics mode" running on memory limited systems should not be further encumbered with yet another FBO.

10110111 commented:
> This architecture should be described somewhere!!!

Well, it's still gonna change when I split the rendering between layers to make labels etc. separate from physical objects and tone mapping.

> Dithering should be applied only to the final view

This must always be true, since it's a conversion from full color depth to the monitor space.

> If possible, gridlines or DSO markers, or even the MilkyWay/ZL could be made visible again without streaking. (I.e., not put into that "celestial" FBO. (But another FBO for MW/ZL/DSO textures/DSS/HiPS is yet another indirection, may be overkill, so IMHO it's OK to just switch them off).

What would a star streak represent if the Milky Way remains at one position? Sounds unphysical. If you want to simulate a long-exposure photo, all celestial objects should move and be smeared. Otherwise you could just as well draw the stars as lines (similar to orbits).

> The "low graphics mode" running on memory limited systems should not be further encumbered with yet another FBO.

The low graphics mode shouldn't receive any new graphical features IMO. Only fix the existing bugs (except the ones requiring a serious rework).

gzotti commented:
> > This architecture should be described somewhere!!!
> 
> Well, it's still gonna change when I split the rendering between layers to make labels etc. separate from physical objects and tone mapping.

Yes, but just as code evolves, docs can evolve and later versions can supersede current docs. If somebody wants to join right now, it's super hard without some information on what goes where. (It has always been like that, though...) 
 
> > Dithering should be applied only to the final view
> This must always be true, since it's a conversion from full color depth to the monitor space.

Sure. Just in the current state, it accumulates, so I must switch it off.

 
> > If possible, gridlines or DSO markers, or even the MilkyWay/ZL could be made visible again without streaking. (I.e., not put into that "celestial" FBO. (But another FBO for MW/ZL/DSO textures/DSS/HiPS is yet another indirection, may be overkill, so IMHO it's OK to just switch them off).
> 
> What would a star streak represent if the Milky Way remains at one position? Sounds unphysical. If you want to simulate a long-exposure photo, all celestial objects should move and be smeared. Otherwise you could just as well draw the stars as lines (similar to orbits).

The MW and other extended features could either continue rotating with the stars as they are newly plotted, or stop at the place when this mode was activated, or be smeared. In the last case, it must dimmed down however, or the sky will be white within seconds. Therefore I currently switch it completely off.

> 
> > The "low graphics mode" running on memory limited systems should not be further encumbered with yet another FBO.
> 
> The low graphics mode shouldn't receive any new graphical features IMO. Only fix the existing bugs (except the ones requiring a serious rework).

OK for me.

dromeuf commented:
Hi, I'm looking to make an illustration with a long exposure effect of circumpolar star trails. Does this feature work in Stellarium 24.2 ? In the documentation I can only find the trail planets. Kind regards,

dromeuf commented:
Georges, I did read the thread you created in 2021. I didn't know whether you had completed this experiment. Thank you for all your great work on the Stellarium project.

dromeuf commented:
Further to our discussion, here is the solution that I found last night to illustrate the circumpolar rotation at the northern horizon of the site studied and the movement in the sky of the star Vega in 4000 BC which rubs the horizon of the place of observation :
- stop animation
- set the date
- capture screen with capture tools
- change the date (min)
- capture screen.... n times...
- Use StarStaX https://markus-enzweiler.de/software/starstax/ for stacking capture screen

![IllustrationStarsTrailsStellarium](https://github.com/user-attachments/assets/57390057-a40c-410e-b4e2-8d239b2294ee)

I read in the latest Stellarium 24.3 user guide that the representation of the proper motion is good to within 100,000 years but that the ‘Carte du Ciel’ software will be more accurate than Stellarium. In 2021 Georg you indicated in #1546 + or - 40,000 years. By comparing with ‘Carte du Ciel’ on the star Vega, I have the impression that the proper motions of Stellarium are those of the Hipparcos 2007 catalogue. In 6,000 years, Vega's proper motion is 35'. When I calculate the angular distance between the coordinates for the date between Stellarium and ‘Carte du ciel’, I find a difference of about 4'. I don't have the explanation because I haven't had time to examine the details of the calculations made by the 2 software packages: precession model, mean coordinates, etc.

gzotti commented:
I did not intend to say Stellarium is accurate within &pm;40.000 years. The settable date range is &pm;100.000 just to have round numbers. The available models concentrate on present times. A few like precession declare date range (exceeding the 100.000), no information is given for nutation, so we invented the limit given in the Guide. VSOP87 is recommended for -4000...+8000, the DE[34]31 solutions -13.000...+17.000. It should be clear that linear proper motion is not enough for accurate positions over tens of thousands of years, still the _idea_ of motion is often more important than the actual figure change. Just don't trust it to the last decimal place! (Don't ask us to reduce the number of displayed digits in prehistory!)  And in a previous (rejected) PR a suboptimal way to fix the current problem can still be seen. 

Currently a new star catalog is in preparation (#3982). therefore I will not compare anything that involves the current, known-to-be-somewhat-flawed solution.

dromeuf commented:
In my case and for the purpose mentioned, the precision is sufficient for visual illustration. Thanks again and congratulations on all the work done by the Stellarium team. It's also good news if your synergy focuses on updating star catalogs. I've read #3982

gzotti committed:
Do not repeat drawing constellations and asterisms in no-clear mode

gzotti committed:
Keep landscape unilluminated in no-clear mode
- also avoid drawing lightscape and fog layers
- also avoid special markers (esp. compass marks)

gzotti committed:
Don't draw hint/label in no-clear mode

gzotti committed:
Reduce star size and streak brightness

gzotti committed:
Reset sky drawing after movement

gzotti commented:
We're getting somewhere! Run, switch off atmosphere, set some timelapse and press Ctrl-Alt-C. 
Not sure whether I should remove the attempted ViewportEffect.

gzotti committed:
Reset no-clear mode when time is changed

gzotti committed:
Avoid ArchaeoLines redraw in no-clear mode

gzotti committed:
Freeze atmosphere and draw stars in the daytime sky
- This fails sometimes, brightening the sky to bright blue or even white-out.

gzotti commented:
The basic appearance is to freeze the last frame (including lines, Milky Way, even atmosphere, and add star streaks as if there was no atmosphere.  I think it has its applications in demos or teaching.  

Technically, this is now continuing my first attempt of "forgetting" the framebuffer reset. However, to allow a nicely filled background, one "regular" frame is filled, and only starting with the next draw cycle the "large" modules (MilkyWay, Zodiacal Light, gridlines, etc.) are suppressed. Any movement, zoom or time jump resets the scene. 

Closing Dialogs should also reset the scene, but only Search dialog seems to behave. Not sure about the others.
EDIT: Solved.

gzotti committed:
Reset scene with location change

gzotti committed:
Improve sky model suppression in no-clear mode

gzotti committed:
Improve sky suppression

gzotti committed:
Reset scene on dialog closing.

gzotti committed:
Scene reset at window resize

gzotti committed:
SUG: Describe new mode

gzotti committed:
Rename action

gzotti committed:
Disable draw in relevant plugins in no-clear mode

- Adjust Satellites drawing: suppress orbits, shadow circles etc.

gzotti commented:
Any comments here? I'd probably take away Guillaume's Viewport Effect as I have not continued that path. Else, it works well for me. IMHO this is a feature best used by instructors either demonstrating daily motion, or just as closing show effect. By default you cannot accidentally trigger it but have to configure your hotkey.

alex-w commented:
> Any comments here? I'd probably take away Guillaume's Viewport Effect as I have not continued that path. Else, it works well for me. 

OK

> IMHO this is a feature best used by instructors either demonstrating daily motion, or just as closing show effect. By default you cannot accidentally trigger it but have to configure your hotkey.

Hmm... what about some "presets", like disabling atmosphere and SSO, when star trails effect is enabled?

gzotti commented:
Try it. As described above, Atmosphere, Milky Way, DSO images are rendered just once, after that only stars and SSO are rendered and accumulate the long-time exposure. This avoids a white-out that else happens within seconds.

alex-w commented:
> Try it. As described above, Atmosphere, Milky Way, DSO images are rendered just once, after that only stars and SSO are rendered and accumulate the long-time exposure. This avoids a white-out that else happens within seconds.

At the moment it draw the trails for artificial satellites too - IMHO this shouldn't be visible to users.

gzotti commented:
You can actually nicely illustrate the nightmare for photographers.
<img width="2272" height="1290" alt="Screenshot (12)" src="https://github.com/user-attachments/assets/60241151-251a-4924-8ef4-5af17fc9938b" />

For Satellite icons and orbit/shadows: indeed these should be suppressed.

Meh - disabling icons with the switchover is not possible, a first image would show them. Switching off icons is left to the operator.

10110111 commented:
In star trails mode the bottom of the Guereins landscape gets overexposed.

gzotti commented:
Ah, thanks for catching this. I never use this but sure, old_style needs to work.

gzotti committed:
fix drawing old_style bottom panel

gzotti committed:
Move draw exclusions back to module's own draw()

gzotti committed:
Amend previous

10110111 commented:
Generally, the implementation seems to be too invasive. Can't we just draw the few items that we want to accumulate after the first frame containing everything has been drawn? I.e. decide whether we want to draw landscape, grids, etc. at the level of module choice, rather than put dozens of `if(flagClearSky)` everywhere?

gzotti commented:
Well, I just reverted this "global" selection in StelApp in the "forall(modules):draw()" loops to not have too many exception paths and have one common pattern: a flag check at begin of each module's draw().  

It turned out it has to be more invasive than wanted in 2021, but that's the difference between "I'd like to have" and "what has to be done for it" (to avoid ugliness).

I can re-enable the draw exclusion in StelApp based on module name string comparison to silence most affected modules if you think this is the better way, but I fear that's slower than checking a Boolean flag.  For stars and planets with all their labeling it remains more complicated in any case.

10110111 commented:
No, the hybrid way (as was before your last commits) is of course even
worse than what we have now.
My idea was not to filter, but rather to only call the modules we are
interested in. But I guess the current intermixture of the scene and labels
doesn't let us do it elegantly...

On Sat, 16 May 2026, 19:12 Georg Zotti, ***@***.***> wrote:

> *gzotti* left a comment (Stellarium/stellarium#1888)
> <https://github.com/Stellarium/stellarium/pull/1888#issuecomment-4466673539>
>
> Well, I just reverted this "global" selection in StelApp in the
> "forall(modules):draw()" loops to not have too many exception paths and
> have one common pattern: a flag check at begin of each module's draw().
>
> It turned out it has to be more invasive than wanted in 2021, but that's
> the difference between "I'd like to have" and "what has to be done for it"
> (to avoid ugliness).
>
> I can re-enable the draw exclusion in StelApp based on module name string
> comparison to silence most affected modules if you think this is the better
> way, but I fear that's slower than checking a Boolean flag. For stars and
> planets with all their labeling it remains more complicated in any case.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/Stellarium/stellarium/pull/1888#issuecomment-4466673539>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABQU3MTUMOHNAOGCDVIEOW343BEJZAVCNFSM6AAAAACYLKQSGOVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHM2DINRWGY3TGNJTHE>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
> You are receiving this because you commented.Message ID:
> ***@***.***>
>

gzotti commented:
So, do you have a better solution for now? In 2023 you proposed changing the overall architecture. This has not happened, so I continued like this.

10110111 commented:
> In 2023 you proposed changing the overall architecture.

Oh yes I remember. I still want to do it, but have lots of other things to do too.

> So, do you have a better solution for now?

Let me look at your implementation for a bit. If I don't come up with anything better, I guess it will do for now.

BTW, do you intend to squash all these commits? There are many fail-and-fix cases here that would better be squashed together.

10110111 commented:
Somehow the bottom part of the old-style landscape is transparent for the Moon and the Sun:

<img width="1920" height="1044" alt="Screenshot_2026-05-16_22-23-51" src="https://github.com/user-attachments/assets/121d420b-4bf7-4ce3-92b6-20e6dfe11e68" />

Internally-created screenshots also don't behave as one might expect:

<img width="1920" height="1044" alt="stellarium-020" src="https://github.com/user-attachments/assets/f68caa7d-ad8a-42dc-a2f4-b40c1d0be9f1" />

gzotti commented:
Actually I think this can be squashed into one new minor feature. 
Landscapes: Oops... will have a look now.
Screenshots: SUG 6.5.2, last sentence. If you find an easy way to do that, fine.

gzotti commented:
Somehow I don't find the decisive difference in the shader code between decor and bottom parts. What works for the sides does not work at the bottom panel! I can only make it work in lowGL mode which also looks OK for me.

gzotti committed:
Fix old_style landscape
- preliminary fix! something is weird with the landscape ground not blocking satellites and subhorizon sun.
- current solution switches to drawLowGL which works.

gzotti commented:
@10110111 Are you OK with switching to lowGL, or can you identify the problem?
(With the lowGL solution, I cold revert the shading var tweaks.)

10110111 committed:
Revert "Fix old_style landscape"

This reverts commit f152e8e499ca47bd09f2cd50427c67094c965af4.

10110111 committed:
Use the same logic for ground as for decor

This fixes growing exposure of the ground (when we remove the early
return) as well as its being transparent to Sun and Moon.

10110111 commented:
So I have fixed the ground issue now.

There are some glitches when panning around during daytime with star trails enabled: the atmosphere flickers, getting sometimes overexposed, other times completely absent. On disabling star trails the atmosphere doesn't immediately reappear for some reason. (I'm using ShowMySky with `atmosphere_resolution_reduction` enabled if this matters).

Generally, the current implementation is hard to follow. Not least because of the confusing name of the flag, but also because it's hard to even tell which objects are supposed to be smeared over the screen. (Is it only stars, planets, and satellites?)

Maybe a better way would be to do as follows:

* Prepare a framebuffer for accumulation of the items we want to see smeared.
* Make (and save) a list of pointers to the modules that should appear in this framebuffer (extracted from the list with the right call order).
* Make (and save) two lists of the remaining modules (first for the ones before the smeared modules, second for those after).
* When rendering:
  - draw the first batch of modules (those before the smeared ones)
  - switch FBO to our accumulation buffer
  - draw the modules we want to smear
  - switch back to the target FBO
  - blit the accumulation buffer
  - draw the rest of modules
* On panning clear the accumulation buffer.
* On screen resize also resize and clear the accumulation buffer.

Of course, the accumulation buffer doesn't have to be created and managed until the user enabled star trails.

This way only the handful of smeared modules will have to deal with the star-trail mode (disabling their labels), the rest will work just as usual (except maybe for the atmosphere that we may want to suppress in this mode).

As a bonus, we'll get back screenshots (except those with a custom size), and avoid the GUI artifacts (currently the bottom panel with the time rate button leaves trails too).

gzotti commented:
> So I have fixed the ground issue now.

Many thanks. I don't know where blindness struck me. 

> There are some glitches when panning around during daytime with star trails enabled: the atmosphere flickers, getting sometimes overexposed, other times completely absent. On disabling star trails the atmosphere doesn't immediately reappear for some reason. (I'm using ShowMySky with `atmosphere_resolution_reduction` enabled if this matters).

Yes, admittedly it's not perfect, just good enough for some effects. I do not understand why the atmosphere sometimes accumulates 2-3 frames (running into white-out) before being silenced, and why it then sometimes remains off. 
 
> Generally, the current implementation is hard to follow. Not least because of the confusing name of the flag, but also because it's hard to even tell which objects are supposed to be smeared over the screen. (Is it only stars, planets, and satellites?)

The name just describes what happens technically, not why. I hope the documentation string is complete enough for understanding. 

Everything that causes large-area brightening must be disabled or the screen turns white within seconds. In 2021-23 my idea was to dim down the accumulated image to limit brightness, which would have also meant limited trail length. Guillaume's filter effect went into this direction but also blurred the view.  In the end only stars (and other star-like extras: Novae, Supernovae), planets, satellites remained. But the other modules should be visible at the onset of the drawing, and the landscape must still cover the ground (if enabled). 


> Maybe a better way would be to do as follows:
> [...]

I can roughly follow this in theory, but it vastly exceeds my OpenGL proficiency, sorry. I wanted to have this feature finally available showing something not the least for satellite annoyance awareness. 

> This way only the handful of smeared modules will have to deal with the star-trail mode (disabling their labels), the rest will work just as usual (except maybe for the atmosphere that we may want to suppress in this mode).

As said, the other modules including atmosphere (awesome twilight composites!) should draw one initial snapshot. E.g., constellations can show stick figures from which star lines grow. Of course, with your mode it would mix "current" view and recorded trails, that's another mode that promises to look interesting. Can we think of this as future version 2 of this feature?
 
> As a bonus, we'll get back screenshots (except those with a custom size), and avoid the GUI artifacts (currently the bottom panel with the time rate button leaves trails too).

I keep the menus visible or switch them off entirely (Ctrl-T). Pressing Ctrl resets the scene.

10110111 committed:
Simplify early returns; fix indentation

10110111 commented:
I don't quite get the logic of sky clearing. Why is there this magic number 3? Why not any other? Besides, `flagClearSkyOnce` is not actually a flag, since it ranges from 0 to 3.

And, I think `flagClearSky` should be renamed to something like `isAccumulatingStarTrails` (with inverse condition), because it's not about sky clearing generally, but rather a hack exclusively for star trails (given that only a few modules actually draw something when this flag is reset).

gzotti commented:
The naming grew historically. I thought a second flag would do for resetting, but it showed I need a 3-2-1-0 countdown to handle the initial 1-shot drawing completely. So, the counter could be renamed, ok. How does `frameClearingResetCounter // for Star Trail Effect ` sound? 
The flagClearSky, again, was the intention in 2021 to signal that at frame start the frame should be reset, as normal case. When disabled, frame is not reset, unless (flag)ClearSkyOnce is not 0, in which case there is some localized logic to handle the reset sequence.   If you think this should be renamed in some inverted sense and handled with inverted logic to be better readable, OK for me.

10110111 commented:
But still, can you explain the idea behind this 3-2-1-0 sequence? Why can't this be done in only two stages (everything once, then accumulation of the few items)?

gzotti commented:
Reducing (Starting) the counter back to 2 causes overly frequent white-out with the daylight sky when pressing Ctrl to reset the trails.

alex-w commented:
@gzotti @10110111 any news?

gzotti commented:
Sorry, conference week. It is as described in the User Guide a hackish approach, but it works usually. It will probably only be found by users that stumble over it while reading the advanced options in the SUG. Separating that into several buffers may be the technically cleaner way, but is beyond my knowledge. I would like this showpiece effect "in", obviously accepting improvements if e.g. @10110111 finds time. I could also add yet another manually modified config option to hide the feature for beginners.

10110111 commented:
If I don't find time to add any improvements here by the end of the next week, I guess this can then go in as is. Otherwise it'll have too small time to be tested before release.

gzotti commented:
The occasional atmo white-out or absence may be caused by the overall brightness estimator getting confused. Must have another look. Weekend, hopefully...

gzotti committed:
Minor fixes

- renamed counter
- clearer docs
- SUG amendments

Signed-off-by: Georg Zotti <Georg.Zotti@univie.ac.at>

10110111 commented:
This crashes unconditionally in debug mode with assertion failure "singleton" in `StelApp.hpp:132`:

```
#8  0x00007ffff4ec1366 in QMessageLogger::fatal
    (this=<optimised out>, msg=0x7ffff522aad8 "ASSERT: \"%s\" in file %s, line %d")
    at /usr/src/qt6-base-6.9.2+dfsg-1ubuntu1/src/corelib/global/qlogging.cpp:883
#9  0x00007ffff4ebf450 in qt_assert
    (assertion=<optimised out>, file=<optimised out>, line=<optimised out>)
    at /usr/src/qt6-base-6.9.2+dfsg-1ubuntu1/src/corelib/global/qassert.cpp:105
#10 0x0000555555fca49f in StelApp::getInstance ()
    at /home/ruslan/src/stellarium/src/core/StelApp.hpp:132
#11 0x000055555644053f in StelRootItem::setSize (this=0x555558afbdd0, size=...)
    at /home/ruslan/src/stellarium/src/StelMainView.cpp:381
#12 0x000055555642fa99 in StelMainView::resizeEvent (this=0x7fffffffd740, event=0x7fffffffcee0)
    at /home/ruslan/src/stellarium/src/StelMainView.cpp:725
#13 0x00007ffff6df750e in QWidget::event (this=0x7fffffffd740, event=0x7fffffffcee0)
    at /usr/src/qt6-base-6.9.2+dfsg-1ubuntu1/src/widgets/kernel/qwidget.cpp:9153
#14 0x00007ffff6e5d866 in QFrame::event (this=0x7fffffffd740, e=0x7fffffffcee0)
    at /usr/src/qt6-base-6.9.2+dfsg-1ubuntu1/src/widgets/widgets/qframe.cpp:521
#15 0x00007ffff4f878da in QCoreApplicationPrivate::sendThroughObjectEventFilters
    (receiver=receiver@entry=0x555558a518a0, event=event@entry=0x7fffffffcee0)
    at /usr/src/qt6-base-6.9.2+dfsg-1ubuntu1/src/corelib/kernel/qcoreapplication.cpp:1243
#16 0x00007ffff6d9d48f in QApplicationPrivate::notify_helper
    (this=<optimised out>, receiver=0x555558a518a0, e=0x7fffffffcee0)
    at /usr/src/qt6-base-6.9.2+dfsg-1ubuntu1/src/widgets/kernel/qapplication.cpp:3294
#17 0x00007ffff4f8add8 in QCoreApplication::notifyInternal2
    (receiver=0x555558a518a0, event=0x7fffffffcee0)
    at /usr/src/qt6-base-6.9.2+dfsg-1ubuntu1/src/corelib/kernel/qcoreapplication.cpp:1106
#18 0x00007ffff4f8ae1d in QCoreApplication::sendEvent
    (receiver=<optimised out>, event=<optimised out>)
    at /usr/src/qt6-base-6.9.2+dfsg-1ubuntu1/src/corelib/kernel/qcoreapplication.cpp:1546
#19 0x00007ffff6df0ebe in QWidgetPrivate::sendPendingMoveAndResizeEvents
    (this=this@entry=0x5555589a6360, recursive=recursive@entry=false, disableUpdates=<optimised out>, disableUpdates@entry=false)
    at /usr/src/qt6-base-6.9.2+dfsg-1ubuntu1/src/widgets/kernel/qwidget.cpp:7982
#20 0x00007ffff6df4ab7 in QWidgetPrivate::show_helper (this=this@entry=0x5555589a6360)
    at /usr/src/qt6-base-6.9.2+dfsg-1ubuntu1/src/widgets/kernel/qwidget.cpp:8033
#21 0x00007ffff6df55c3 in QWidgetPrivate::setVisible (this=0x5555589a6360, visible=<optimised out>)
    at /usr/src/qt6-base-6.9.2+dfsg-1ubuntu1/src/widgets/kernel/qwidget.cpp:8394
#22 0x00007ffff6df4f9e in QWidgetPrivate::showChildren
    (this=<optimised out>, spontaneous=<optimised out>)
    at /usr/src/qt6-base-6.9.2+dfsg-1ubuntu1/src/widgets/kernel/qwidget.cpp:8482
#23 0x00007ffff6df4ad3 in QWidgetPrivate::show_helper (this=this@entry=0x555558d17910)
    at /usr/src/qt6-base-6.9.2+dfsg-1ubuntu1/src/widgets/kernel/qwidget.cpp:8039
#24 0x00007ffff6df55c3 in QWidgetPrivate::setVisible (this=0x555558d17910, visible=<optimised out>)
    at /usr/src/qt6-base-6.9.2+dfsg-1ubuntu1/src/widgets/kernel/qwidget.cpp:8394
#25 0x00007ffff6de337b in QWidget::showFullScreen (this=0x7fffffffd740)
    at /usr/src/qt6-base-6.9.2+dfsg-1ubuntu1/src/widgets/kernel/qwidget.cpp:3056
#26 0x0000555556438136 in StelMainView::setFullScreen (this=0x7fffffffd740, b=true)
    at /home/ruslan/src/stellarium/src/StelMainView.cpp:1500
#27 0x0000555555fc65f0 in main (argc=1, argv=0x7fffffffdaf8)
    at /home/ruslan/src/stellarium/src/main.cpp:477
```

10110111 commented:
> This crashes unconditionally in debug mode

Fixed by 14c74c47221a9cc43d7d25671854247871bb9fe9.

gzotti commented:
Thanks!
